patches/tModLoader/Terraria/ModLoader/BuilderToggle.cs

```diff
@@ -75,13 +86,87 @@ public abstract class BuilderToggle : ModTexturedType, ILocalizedModType
 	/// </summary>
 	public abstract string DisplayValue();
 
+	/// <summary>
+	/// This allows you to change basic drawing parameters or to override the vanillia drawing completely.<para/>
+	/// This is for the icon itself. See <see cref="DrawHover"/> if you want to modify icon hover drawing<para/>
+	/// Return false to stop vanilla drawing code from running. Returns true by default.
+	/// </summary>
+	/// <param name="spriteBatch">The spritebatch to draw on</param>
+	/// <param name="texture">The icon texture</param>
+	/// <param name="position">The position</param>
+	/// <param name="frame">The frame rectangle (aka source rectangle)</param>
+	/// <param name="color">The color the icon is drawn in. Modifications in <see cref="DisplayColorTexture"/> is applied to this value</param>
+	/// <param name="scale">The scale of the icon</param>
+	/// <param name="spriteEffects">The spriteEffects</param>
+	/// <returns>Whether to run vanilla icon drawing code</returns>
+	public virtual bool Draw(SpriteBatch spriteBatch, ref Texture2D texture, ref Vector2 position, ref Rectangle frame, ref Color color, ref float scale, ref SpriteEffects spriteEffects) => true;
+
```

Comment:
`Draw` & `DrawHover` should utilize a structure alike `BossBarDrawParams`, because it's frequent for hooks of this kind to be expanded upon in the future, and changing hook signatures is API-breaking, while adding new fields to structures is completely safe in managed languages. In a perfect world, all TML hooks would've had their own structures.